appcues/src/main/java/com/appcues/action/ActionProcessor.kt

```diff
@@ -43,4 +61,30 @@ internal class ActionProcessor(override val scope: Scope) : KoinScopeComponent {
             currentQueue
         }
     }
+
+    private fun getStepInteractionAction(
```

Comment:
this b asically looks for the "last" valid MetadataSettingsAction implementation so we know we have a step_interaction.

most of the time there will always be at least one, for instance the ContinueAction, LinkAction are both MetadataSettingsAction implementations, but I set this to return a nullable to safe guard in case there isnt any.